### PR TITLE
fix #9919 - prevent "resize" cursor showing for edges of non-resizable dockable widgets

### LINKED ISSUE #9919: [MU4 Issue] Mouse cursor should not change when hover over boundaries that can't be resized
https://user-images.githubusercontent.com/90187801/143875079-33063417-4059-4ead-9186-475983d88780.mp4

**Desktop:**
Win/Mac/LinuxMint

### PR DESCRIPTION
**Pull request by wizofaus**
Resolves: [*(direct link to the issue)*](https://github.com/musescore/MuseScore/issues/9919)

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by wizofaus** [comment]
Fix isn't quite right, doesn't work after undocking and redocking

**Review by RomanPudashkin — APPROVED** [review]

File: src/appshell/qml/dockwindow/DockSeparator.qml [hunk 1]
```diff
@@ -39,8 +39,8 @@ Rectangle {
         anchors.fill: parent
         anchors.margins: -5 //! NOTE: extra space for user convenience
 
-        cursorShape: separatorCpp ? (separatorCpp.isVertical ? Qt.SizeVerCursor : Qt.SizeHorCursor)
-                                   : Qt.SizeHorCursor
+        cursorShape: (separatorCpp && separatorCpp.showResizeCursor) ? (separatorCpp.isVertical ? Qt.SizeVerCursor : Qt.SizeHorCursor)
+                                   : Qt.SizeArrowCursor
```
Comment (cbjeukendrup) [inline src/appshell/qml/dockwindow/DockSeparator.qml:43]: Is `Qt.SizeArrowCursor` correct? Didn't you mean `Qt.ArrowCursor`? (https://doc.qt.io/qt-6/qt.html#CursorShape-enum)

Comment (wizofaus) [reply src/appshell/qml/dockwindow/DockSeparator.qml:43]: Interesting, it must just default to ArrowCursor if you use an undefined type! But yes, obviously a typo.

**Comment by DmitryArefiev** [comment]
Tested #9919 on Win10, Mac12, LinuxMint 20.2 - **FIXED**

**Pull request merged**